roaring.go

```diff
@@ -53,6 +53,53 @@ func (rb *Bitmap) ToBytes() ([]byte, error) {
 	return rb.highlowcontainer.toBytes()
 }
 
+// Modified FNV1a, works on shorts rather than bytes
+func (rb *Bitmap) Checksum() uint64 {
+	const (
+		offset = 14695981039346656037
+		prime = 1099511628211
+	)
+
+	hash := uint64(offset)
+
+	for _, k := range rb.highlowcontainer.keys {
+		hash ^= uint64(k)
+		hash *= prime
+	}
+
+	for _, c := range rb.highlowcontainer.containers {
+		// 0 separator
+		hash ^= 0
+		hash *= prime
+
+		switch c := c.(type) {
+		case *bitmapContainer:
+			iterator := bitmapContainerShortIterator{c, c.NextSetBit(0)}
+			for iterator.hasNext() {
+				hash ^= uint64(iterator.next())
+				hash *= prime
+			}
+		case *arrayContainer:
+			for _, x := range c.content {
+				hash ^= uint64(x)
+				hash *= prime
+			}
+		case *runContainer16:
+			for _, run := range c.iv {
+				for i := uint16(0); i <= run.length; i++ {
```

Comment:
This runs in time proportional to the number of elements, and not proportional to the size of the container. So if you have full container, this will be very slow.

Reply:
Yes, indeed. But if it doesn't, then compressed bitmaps will hash differently than uncompressed bitmaps. I couldn't figure out an alternative.

Reply:
I mean, we could try and detect runs during iteration of the other bitmaps instead, and always hash as if they were compressed. But I'm unsure if that's a good idea. What do you think?

Reply:
Pushed a version doing exactly this.

Reply:
It may be simpler with loops inside loops, but I preferred quick and dirty for the draft.

Reply:
I dislike your fix. I preferred what you had before. I am working on a fix of my own that would be nicer. Give me a few minutes.

Reply:
OK. I can revert those two commits then.